website_sale_product_display_name/views/website_sale_template.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+    Copyright 2016-2017 LasLabs Inc.
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+-->
+
+<odoo>
+
+    <template id="products_item" inherit_id="website_sale.products_item" name="Products item display name">
+        <xpath expr="//span[@t-field='product.image']" position="attributes">
+            <attribute name="t-att-alt" t-value="product.display_name" />
+        </xpath>
+        <xpath expr="//a[@t-field='product.name']" position="replace">
+            <a itemprop="name"
+               t-att-href="keep('/shop/product/%s' % slug(product), page=(pager['page']['num'] if pager['page']['num']>1 else None))"
+               t-field="product.display_name"
+               />
+        </xpath>
+    </template>
+
+    <template id="product" inherit_id="website_sale.product" name="Product display name">
+        <xpath expr="//t[@t-value='product.name']" position="replace">
+            <t t-set="additional_title" t-value="product.display_name"/>
+        </xpath>
+        <xpath expr="//span[@t-field='product.name']" position="replace">
+            <span t-field="product.display_name"/>
+        </xpath>
+        <xpath expr="//span[@t-field='product.image']" position="attributes">
+            <attribute name="t-field-options">
+                {"widget": "image", "class": "product_detail_img", "alt-field": "display_name"}
```

Comment:
This would benefit from https://github.com/OCA/server-tools/tree/9.0/base_view_inheritance_extension, sad it's not in 10.0 yet.